web-wallet/src/lib/containers/MigrateContract/MigrateContract.svelte

```diff
@@ -1,133 +1,196 @@
+    },
+    {
+      disabled: false,
+      label: tokens[network]["BEP-20"].name,
+      value: tokens[network]["BEP-20"].name,
+    },
   ];
 
-  /** @type {String} */
-  let selected = tokens.eth.name;
+  /** @type {TokenNames} */
+  let selectedChain = tokens[network]["ERC-20"].name;
 
   /** @type {Boolean} */
   const migrationInProgress = false;
 
-  /** @type {Number} */
-  let connectedWalletBalance = 1;
+  /** @type {string} */
+  let connectedWalletBalance;
 
-  /** @type {Number | undefined} */
+  /** @type {Number} */
   let amount;
 
+  /** @type {import("svelte/store").Writable<Boolean>} */
+  const hasApprovedCoin = writable(false);
+
   /** @type {HTMLInputElement | null} */
   let amountInput;
 
   /** @type {Boolean} */
   let isMigrationInitialized = false;
 
-  /** @type {Number} */
-  const gasFee = 1;
-
-  /** @type {Boolean} */
-  let isMigrationBeingApproved = false;
-
-  /** @type {String}*/
-  const estimatedTime = "45 min";
-
   /** @type {number} */
   let screenWidth = window.innerWidth;
 
-  const { darkMode } = $settingsStore;
-  const minAmount = 1e-18;
+  /** @type {number} */
+  let migrationStep = 0;
 
-  $: ({ address, chainId, isConnected } = $account);
+  let isInputDisabled = false;
 
-  $: maxSpendable = connectedWalletBalance - gasFee;
-  $: isAmountValid =
-    typeof amount === "number"
-      ? amount >= minAmount && amount <= maxSpendable
-      : false;
-  $: ({ language } = $settingsStore);
-  $: duskFormatter = createCurrencyFormatter(language, "DUSK", 9);
+  function toggleDisabledAmount() {
+    isInputDisabled = !isInputDisabled;
+  }
+
+  const minAmount = 1e-9;
 
   /** @param {Number} id */
   async function handleSwitchChain(id) {
     try {
       await switchChain(wagmiConfig, { chainId: id });
     } catch (e) {
-      selected =
-        chainId === tokens.eth.chainId ? tokens.eth.name : tokens.bnb.name;
+      selectedChain =
+        chainId === tokens[network]["ERC-20"].chainId
+          ? tokens[network]["ERC-20"].name
+          : tokens[network]["BEP-20"].name;
     }
   }
 
   // @ts-ignore
-  function onNetworkChange(e) {
+  async function onChainSwitch(e) {
     if (isConnected) {
-      if (e?.target?.value === tokens.bnb.name) {
-        handleSwitchChain(bsc.id);
+      if (e?.target?.value === tokens[network]["BEP-20"].name) {
+        handleSwitchChain(tokens[network]["BEP-20"].chainId);
       } else {
-        handleSwitchChain(mainnet.id);
+        handleSwitchChain(tokens[network]["ERC-20"].chainId);
       }
+
+      await getBalance();
     }
   }
 
-  function switchNetwork() {
+  function switchToSelectedChain() {
     const currentChainId =
-      selected === tokens.eth.name ? tokens.eth.chainId : tokens.bnb.chainId;
+      selectedChain === tokens[network]["ERC-20"].name
+        ? tokens[network]["ERC-20"].chainId
+        : tokens[network]["BEP-20"].chainId;
     if (chainId !== currentChainId) {
       handleSwitchChain(currentChainId);
     }
   }
```

Comment:
I would generally refactor the three function above as follows:

```
  /** @param {Number} chainId */
  async function handleSwitchChain(chainId) {
    try {
      await switchChain(wagmiConfig, { chainId });
    } catch (error) {
      const { ["ERC-20"]: erc20, ["BEP-20"]: bep20 } = tokens[network];
      selectedChain = chainId === erc20.chainId ? erc20.name : bep20.name;
    }
  }

  // @ts-ignore
  async function onChainSwitch(event) {
    if (!isConnected) return;

    const selectedChainName = event.target?.value;
    const { ["ERC-20"]: erc20, ["BEP-20"]: bep20 } = tokens[network];
    const chainIdToSwitch =
      selectedChainName === bep20.name ? bep20.chainId : erc20.chainId;

    await handleSwitchChain(chainIdToSwitch);
    await getBalance();
  }

  function switchToSelectedChain() {
    const { ["ERC-20"]: erc20, ["BEP-20"]: bep20 } = tokens[network];
    const currentChainId =
      selectedChain === erc20.name ? erc20.chainId : bep20.chainId;

    if (chainId !== currentChainId) {
      handleSwitchChain(currentChainId);
    }
  }
```

Also adding jsdoc comments stating that `switchToSelectedChain` is used to switch to the chain as selected in the 3rd party wallet provider, where as the `onChainSwitch` is used when the user uses the `ExclusiveChoice` component. Otherwise these are way too similar in naming and functionality and it takes time to figure out the purpose each function is used for.